### PR TITLE
Moved cache file management to Scene class

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

File: brayns/common/geometry/Cone.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ struct Cone
     {
     }
 
+    Cone() {}
```
Comment (tribal-tec) [inline brayns/common/geometry/Cone.h]: = default

File: brayns/common/scene/Scene.cpp [hunk 2]
```diff
@@ -669,4 +675,325 @@ void Scene::setCylinder(const size_t materialId, const uint64_t index,
...
+    BRAYNS_INFO << "Saving scene to binary file: " << filename << std::endl;
+    std::ofstream file(filename, std::ios::out | std::ios::binary);
+    if (!file.good())
+    {
+        BRAYNS_ERROR << "Could not open cache file " << filename << std::endl;
+        return;
+    }
+
+    const size_t version = CACHE_VERSION;
+    file.write((char*)&version, sizeof(size_t));
```
Comment (tribal-tec) [inline brayns/common/scene/Scene.cpp]: why not file << version? everywhere in this file?

File: brayns/common/scene/Scene.cpp [hunk 3]
```diff
@@ -669,4 +675,325 @@ void Scene::setCylinder(const size_t materialId, const uint64_t index,
...
+    std::ifstream file(filename, std::ios::in | std::ios::binary);
+    if (!file.good())
+    {
+        BRAYNS_ERROR << "Could not open cache file " << filename << std::endl;
+        return;
+    }
+
+    // File version
+    size_t version;
+    file.read((char*)&version, sizeof(size_t));
```
Comment (tribal-tec) [inline brayns/common/scene/Scene.cpp]: also works backwards :) file >> version;

File: brayns/common/scene/Scene.cpp (hunk 2 above)
Comment (favreau) [reply brayns/common/scene/Scene.cpp]: Not all elements have this operator implemented. Some of those elements being in process of being replaced, I will leave your comments to a later PR if that's acceptable?

File: brayns/common/scene/Scene.cpp (hunk 3 above)
Comment (favreau) [reply brayns/common/scene/Scene.cpp]: dito

**Pull request merged**